keyboards/ghost_spot/ghost54/keymaps/via/keymap.c

```diff
@@ -0,0 +1,192 @@
+ *                          `---------------------'           '------''-------------'
+ */
+[3] = LAYOUT(
+  KC_ESC,  _______,   KC_F10, KC_F11, KC_F12,_______,              _______, KC_F22, KC_F23, KC_F24, _______, KC_PSCR,
+  KC_TAB,  _______,   KC_F7,  KC_F8,  KC_F9, _______,              _______, KC_F19, KC_F20, KC_F21, _______, _______,
+  KC_LSFT, _______,   KC_F4,  KC_F5,  KC_F6, _______,              _______, KC_F16, KC_F17, KC_F18, _______, _______,
+  KC_LCTL, _______,   KC_F1,  KC_F2,  KC_F3, _______,              _______, KC_F13, KC_F14, KC_F15, _______, _______,
+                   KC_ENT, KC_SPC, MO(1),  LGUI(KC_TAB),         KC_MPLY, OSL(2), KC_BSPC, KC_DEL
+)
+};
+
+/* Behaviour of the ENCODERS  */
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (get_highest_layer(layer_state|default_layer_state) == 0) {
+        if (index == 0) { // Alt+TAB and Alt+Shift+TAB
+            register_code(KC_LALT);
+            is_alt_tab_active = true;
+            if (clockwise) {
+                tap_code(KC_TAB);
+            } else {
+                register_code(KC_LSFT);
+                tap_code(KC_TAB);
+                unregister_code(KC_LSFT); // this fixes the getting stuck problem
+            }
+            alt_tab_timer = timer_read();
+        } else if (index == 1) {
+            if (clockwise) { // Scroll horizontally words
+                tap_code(C(KC_LEFT));
+            } else {
+                tap_code(C(KC_RIGHT));
+            }
+        }
+    } else if (get_highest_layer(layer_state|default_layer_state) == 1) {
+        if (index == 0) { // PGUP and PGDN
+            if (clockwise) {
+                tap_code(KC_PGDN);
+            } else {
+                tap_code(KC_PGUP);
+            }
+        } else if (index == 1) { // Scroll tabs
+            if (clockwise) {
+                tap_code16(C(KC_TAB));
+            } else {
+                tap_code16(S(C(KC_TAB)));
+            }
+        }
+    } else if (get_highest_layer(layer_state|default_layer_state) == 2) {
+        if (index == 0) { // PGUP and PGDN
+            if (clockwise) {
+                tap_code(KC_PGDN);
+            } else {
+                tap_code(KC_PGUP);
+            }
+        } else if (index == 1) { // Scroll tabs
+            if (clockwise) {
+                tap_code16(C(KC_TAB));
+            } else {
+                tap_code16(S(C(KC_TAB)));
+            }
+        }
+    } else if (get_highest_layer(layer_state|default_layer_state) == 3) {
+        if (index == 0) { // History Scrubbing
+            if (clockwise) {
+                tap_code(C(KC_Y));
+            } else {
+                tap_code(C(KC_Z));
+            }
+        } else if (index == 1) {
+            if (clockwise) { // Volume Control
+                tap_code16(KC_VOLU);
+            } else {
+                tap_code16(KC_VOLD);
+            }
+        }
+    }
+    return false;
+};
```

Comment:
This block should be replaced with encoder map, as it handles all of this, already.